surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java

```diff
@@ -185,6 +188,13 @@ private SimpleReportEntry createReportEntry( TestIdentifier testIdentifier,
                 stw, elapsedTime, reason, systemProperties );
     }
 
+    private boolean isFailedContainer( TestIdentifier testIdentifier,
+                                      TestExecutionResult testExecutionResult )
+    {
+        return testIdentifier.isContainer() && testExecutionResult != null
```

Comment:
The call of callback method with `null` in `TestExecutionListener` does not make sense. Eventhough the Javadoc on `TestExecutionListener` does not say anything about passing the null to the method parameter.
Here static import of whole `TestExecutionResult.Status.FAILED` whould make the line brief.

Reply:
I'll switch to a static import, but I do need the null check. The method directly below this one passes a null value for a `testExecutionResult` (which ultimately makes its way over to here): https://github.com/apache/maven-surefire/pull/267/files/38ad13f5a6a628282064d1b0976e1ea69a135b0e#diff-ad07b0c74885ecb3d687dcc85495e390R198-R201

Reply:
I think I see the confusion. I have some changes to how the logic is applied to address some other cases. Those changes also end up reducing the scope of that null check to only the appropriate places. I'm currently waiting for it to do a full build, so I'll probably push them a while later today. Once that's done (assuming it passes), I'll get started on the IT.

Reply:
Alright, can you start writing the IT?

Reply:
Yep, on it

Reply:
IT added for all the functionality changes (plus some additional unit tests for the changes regarding errors vs failures).

Reply:
The PR that you closed in favor of this one, #257, the owner had gotten started on some additional IT's for their own issue:

https://github.com/apache/maven-surefire/compare/master...t-8ch:e6c3c4fef38ee6a5e5352bec447e047713b7cd0b

Should I copy those IT's into this? I'm not super familiar with `@TestFactory` or `@TestTemplate`, and I'm not sure what the state of those IT's are, but would that be preferable as a means to also resolve that other issue? I did a quick local check after my first couple commits to see if this addresses that issue, and if I remember correctly, it does.

Reply:
yes, you can take it as a motivation with one exception. Pls do not reuse the existing directory `junit-platform-engine-jupiter`. Create your own. The reason is that this directory already contains some tests

Reply:
Did I do that? I know the closed PR did that, but I put mine in the surefire-1741 directory like the BeforeAll one did (for its own issue).

Reply:
In any event, IT's added for the other issue. That should be everything.